ResearchKit/Common/ORKFormItemCell.m

```diff
@@ -1148,3 +1156,168 @@ - (BOOL)textField:(UITextField *)textField shouldChangeCharactersInRange:(NSRang
+    
+    [self.contentView addConstraints:constraints];
+}
+
+- (void)applyAnswerFormat {
+    
+}
+
+- (void)setFormItem:(ORKFormItem *)formItem {
+    [super setFormItem:formItem];
+    
+    if (_selectionView) {
+        [_selectionView setPlaceholderText:formItem.placeholder];
+    } else {
+        _placeholder = formItem.placeholder;
+    }
+    
+    [self applyAnswerFormat];
+}
+
+- (void)answerDidChange {
+    id answer = self.answer;
+    
+    if (answer == ORKNullAnswerValue()) {
+        answer = nil;
+    }
+    
+    _selectionView.answer = (MKPlacemark *)answer;
+    [_selectionView setTextColor:[UIColor blackColor]];
+}
+
+- (void)setEditingHighlight:(BOOL)editingHighlight {
+    _editingHighlight = editingHighlight;
+    [_selectionView setTextColor:( _editingHighlight ? [self tintColor] : [UIColor blackColor])];
+}
+
+- (void)locationSelectionViewDidBeginEditing:(ORKLocationSelectionView *)view {
+    self.editingHighlight = YES;
+    [_selectionView showMapView];
+    [self.delegate formItemCellDidBecomeFirstResponder:self];
+}
+
+- (void)locationSelectionViewDidEndEditing:(ORKLocationSelectionView *)view {
+    self.editingHighlight = NO;
+    [self.delegate formItemCellDidResignFirstResponder:self];
+}
+
+- (void)locationSelectionViewDidChange:(ORKLocationSelectionView *)view {
+    [self inputValueDidChange];
+}
+
+- (void)locationSelectionViewNeedsResize:(ORKLocationSelectionView *)view {
+    UITableView *tableView = [self parentTableView];
+    
+    _heightConstraint.constant = _selectionView.intrinsicContentSize.height;
+    
+    [tableView beginUpdates];
+    
+    CGRect convertedVisibleRect = [tableView convertRect:view.bounds fromView:view];
+    [tableView scrollRectToVisible:convertedVisibleRect animated:YES];
+    
+    [tableView endUpdates];
+}
+
+- (void)locationSelectionView:(ORKLocationSelectionView *)view didFailWithError:(NSError *)error {
+    [self showErrorAlertWithTitle:ORKLocalizedString(@"LOCATION_ERROR_TITLE", @"") message:error.localizedDescription];
+}
+
+- (void)inputValueDidChange {
+    [self ork_setAnswer:((_selectionView.answer != nil) ? _selectionView.answer : ORKNullAnswerValue())];
+    [super inputValueDidChange];
+}
+
+- (BOOL)becomeFirstResponder {
+    return [_selectionView becomeFirstResponder];
+}
+
+- (BOOL)resignFirstResponder {
+    BOOL resign = [super resignFirstResponder];
+    resign = [_selectionView resignFirstResponder] || resign;
+    return resign;
+}
+
+- (BOOL)isAnswerValid {
+    id answer = self.answer;
+    
+    if (answer == ORKNullAnswerValue()) {
+        return YES;
+    }
+    
+    ORKAnswerFormat *answerFormat = [self.formItem impliedAnswerFormat];
+    ORKLocationAnswerFormat *locationFormat = (ORKLocationAnswerFormat *)answerFormat;
+    
+    MKPlacemark *placemark = (MKPlacemark *)answer;
+    NSString *string = [placemark ork_stringValue];
+    
+    return [locationFormat isAnswerValidWithString:string];
```

Comment:
Why not directly use `- (BOOL)isAnswerValid:(id)answer;` from `answerFormat` to avoid the conversion between placemark and string?